providers/implementations/kdfs/hkdf.c

```diff
@@ -259,16 +302,29 @@ static int hkdf_common_set_ctx_params(KDF_HKDF *ctx, const OSSL_PARAM params[])
     if (ossl_param_is_empty(params))
         return 1;
 
-    if (OSSL_PARAM_locate_const(params, OSSL_ALG_PARAM_DIGEST) != NULL) {
-        const EVP_MD *md = NULL;
+    if ((p = OSSL_PARAM_locate_const(params, OSSL_ALG_PARAM_DIGEST)) != NULL) {
+        const EVP_MD *md = ossl_prov_digest_md(&ctx->digest);
 
-        if (!ossl_prov_digest_load_from_params(&ctx->digest, params, libctx))
-            return 0;
+        if (ctx->fixed_digest && md != NULL) {
+            char digest[OSSL_MAX_NAME_SIZE];
```

Comment:
It would probably be ok to just error here..
Setting the digest is not supported for these..

Reply:
Other reviewers might argue otherwise..

Reply:
I'd be okay with just erroring out here.